fetch.bs

```diff
@@ -4227,6 +4259,135 @@ prefetch, or to treat it differently when counting page visits.
+ "<code>https</code>", then set |isSecure| to true.
+
+ <li>
+  <p>Let |httpOnlyAllowed| be true.
+
+  <p class=note>True follows from this being invoked from <a>fetch</a>, as opposed to the
+  <code>document.cookie</code> getter steps for instance.
+
+ <li><p>Let |sameSiteStrictOrLaxAllowed| be true if the result of [=determine the same-site mode=]
+ for |request| is "<code>StrictOrLess</code>", and false otherwise.
+
+ <li><p><a for=list>For each</a> <var>header</var> of <var>response</var>'s
+ <a for=response>header list</a>:
+
+ <ol>
+  <li><p>If <var>header</var>'s <a for=header>name</a> is not a <a>byte-case-insensitive</a> match
+  for `<code>Set-Cookie</code>`, <a for=iteration>continue</a>.
+
+  <li><p><a>Parse and store a cookie</a> given <var>header</var>'s <a for=header>value</a>,
+  |isSecure|, <var>request</var>'s <a for=request>current URL</a>'s <a for=url>host</a>,
+  <var>request</var>'s <a for=request>current URL</a>'s <a for=url>path</a>, |httpOnlyAllowed|,
+  |allowNonHostOnlyCookieForPublicSuffix|, and |sameSiteStrictOrLaxAllowed|.
+
+  <li><p><a>Garbage collect cookies</a> given <var>request</var>'s
+  <a for=request>current URL</a>'s <a for=url>host</a>.
+ </ol>
+</ol>
+</div>
+
+<h3 id=cookie-infrastructure>Cookie infrastructure</h3>
+
+<p>These algorithms are not only for use with the `<code>Cookie</code>` header, and are used in
+other specifications.
+
+<div algorithm>
+<p>To <dfn>determine the same-site mode</dfn> for a given <a for=/>request</a> <var>request</var>:
+
+<ol>
+ <li><p><a for=/>Assert</a>: <var>request</var>'s <a for=request>method</a> is "<code>GET</code>"
+ or "<code>POST</code>".
+
+ <li><p>If <var>request</var>'s <a for=request>top-level navigation initiator origin</a> is not
+ null and is not <a for=/>same site</a> to <var>request</var>'s <a for=request>URL</a>'s
+ <a for=url>origin</a>, return "<code>UnsetOrLess</code>".
+
+ <li><p>If <var>request</var>'s <a for=request>method</a> is "<code>GET</code>" and
+ <var>request</var>'s <a for=request>destination</a> is "document", return
+ "<code>LaxOrLess</code>".
+
+ <li><p>If <var>request</var>'s <a for=request>client</a>'s <a for=environment>ancestry</a> is
+ "<code>cross-site</code>", then return "<code>UnsetOrLess</code>".
+
+ <li><p>If <var>request</var>'s <a for=request>redirect-taint</a> is "<code>cross-site</code>",
+ then return "<code>UnsetOrLess</code>".
+
+ <li><p>Return "<code>StrictOrLess</code>".
+</ol>
+</div>
+
+<div algorithm>
+<p> To <dfn export>clear site cookies</dfn> for <a for=/>origin</a> <var>origin</var>, run these steps:
+
+<ol>
+ <li><p>Remove all cookies that were stored with a host is either equal to |origin|'s <a for=url>host</a>,
+ or have a <a>registrable domain</a> equal to |origin|'s <a for=url>host</a>'s <a>registrable domain</a>.
+
+ <p class=XXX>The cookie specification doesn't yet have a way to do this algorithmically, so we
+ settle for this definition.
+</ol>
+</div>
```

Comment:
I've lost track of why we want this in v1. I think if we have something like this we also need something that calls it. Perhaps it's easier after all to leave this as follow-up? Apologies if I was the one pushing for this, but I don't think this is what I had in mind.

Reply:
I wasn't pushing for it- I'll pull this out for now and once we have a caller queued up, we can add it.